<!DOCTYPE HTML PUBLIC "-//W2C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8">

  <style type="text/css">
  ADDRESS {font-family: Arial, Helvetica;}
  BODY {margin-top:5%; margin-left:10%;margin-right:10%;margin-bottom:10%;
  font-family: Arial, Helvetica; line-height:130%}
  H1{line-height:120%}
  TD {font-family: Arial, Helvetica;}
  P {font-family: Arial, Helvetica; text-align: justify;}
  A:link {text-decoration:none;}
  A:vlink {text-decoration:none;}
  </style>
  
  
  <title>BIOS 555: High-throughput data analysis using R and Bioconductor</title>
  </head>
<body width="80%">
<a name="Outline"></a>
<h1><center>BIOS 555: High-throughput data analysis using R and Bioconductor</h1>


<!--
<br><hr size="5" width="100%"><br>
<table width="100%">
  <tbody>
    <tr>
      <th width="20%"><a href="#info">Class Info</a></th>
      <th width="20%"><a href="#announcements">Reading materials</a></th>
      <th width="20%"><a href="#projects">Projects</a></th>
      <th width="20%"><a href="#class">Class notes</a></th>
    </tr>
  </tbody>
</table>
<br><hr size="5" width="100%"><br>

-->
<br><hr size="5" width="100%">
<a name="info"><h3>Class Information</h3></a>

<ul>
  <li><b>Instructor</b>: <a href="http://www.haowulab.org">Hao Wu</a>. 
    Email: hao.wu at emory dot edu.</li>
  <li><b>TA</b>: Luxiao Chen (email: luxiao.chen at emory dot edu)</li>
  <li><b>Class time and location</b>: Aug 25-Oct 18 2021, Monday and Wednesday 3:10-5:00PM at <b>GCR 115</b>.
	No class on Sep 6th (Labor day) and Oct 11 (fall break).
<!--  <li><b>Online access</b>:  <a href="https://zoom.us/j/8294730806">https://zoom.us/j/8294730806</a>.-->
  <li><b>TA office hour</b>: Make appointment, or email question. 
  <li> <b>Grading</b>: homeworks, class participation and final project. 
</ul>

<center><hr size="5" width="100%"></center>

<a name="info"><h3>Summary</h3>
This course covers the basics of 
analyses of genomic data from high-throughput technologies, 
mainly microarray and second-generation sequencing.
Topics include the biological motivations, experimental procedures
and statistical methods for different technologies. 
Usage of existing software packages (mainly in R/Bioconductor) 
for analyzing various genomic data will be introduced.
<p>
This class put more emphases on applications instead of 
statisical theories. Upon completion of the class, students 
are expected to be able to:
<ol>
<li>Understand the biological motivations and technological procedures of 
high-throughput experiments including different types of microarrays and second generation sequencing.
<li>Understand statistical challenges and existing methods for analyzing the data 
generated from high-throughput experiments.
<li>Analyze high-throughput data using R/Bioconductor and other open source software.
</ol>

<b>Prerequisite</b>:
BIOS 501 or equivalents. Basic programming experiences in R. 
Some experience in command line driven OS (DOS or Linux).
<p>

<b>Grading</b>:
Four sets of homework, each worth 15%. Final project 30%. Class participate 10%.
<p>

<b>Lab</b>: Bring laptop to the labs.

<center><hr size="5" width="100%"></center>

<!-- announcements -->
<a name="announcements"><h3>Reading Materials:</h3> </a><p>
Here are some reading materials related to the class. It is NOT required  
to read them all, but it'll help you get better understanding of certain materials. 
<ol>
<li>R programming: <a href="http://cran.r-project.org/doc/contrib/Paradis-rdebuts_en.pdf">
R for beginners</a> is highly recommended. 
<li>Bioconductor. There are several books listed on the Bioconductor website at 
<a href="http://www.bioconductor.org/help/publications/">here</a>, but they are 
pretty outdated. The best way to learn is to read the manual or "vignette"
for each package. 
<li>Important papers:
  <ul>
    <li>Gene expression microarray: 
      <a href="http://www.nature.com/nrg/journal/v7/n1/full/nrg1749.html">A nice review article</a>,
      <a href="rma.pdf">RMA</a>, 
      <a href="gcrma.pdf">GCRMA</a>, <a href="sam.pdf">SAM t-test</a>,
      <a href="limma.pdf">limma</a>.
    <li>Second generation sequencing:
      <a href="sgseq-review.pdf">A review article by Shendure and Ji</a>,
      <a href="http://genomebiology.com/2009/10/3/R25">bowtie paper</a>, 
    <li>RNA-seq: 
      <a href="http://genomebiology.com/2010/11/10/R106">DEseq</a>, 
      <a href="http://bioinformatics.oxfordjournals.org/content/23/21/2881.abstract">edgeR</a>,
      <a href="http://bioinformatics.oxfordjournals.org/content/25/8/1026.abstract">Jiang et al. for isoform expression</a>, 
      <a href="http://www.nature.com/nbt/journal/v28/n5/full/nbt.1621.html">cufflink</a>,
      <a href="http://bioinformatics.oxfordjournals.org/content/25/9/1105.abstract">Tophat</a>.
    <li>ChIP-seq: <a href="http://genomebiology.com/2008/9/9/R137">MACS</a>, 
      <a href="http://www.nature.com/nbt/journal/v26/n11/abs/nbt.1505.html">CisGenome</a>, 
      <a href="http://www.nature.com/nbt/journal/v27/n1/abs/nbt.1518.html">PeakSeq</a>,
      <a href="http://www.nature.com/nmeth/journal/v5/n9/full/nmeth.1246.html">QuEST</a>.
    <li>BS-seq: 
        <a href="http://genomebiology.biomedcentral.com/articles/10.1186/gb-2012-13-10-r83">BSmooth</a>,
        <a href="http://nar.oxfordjournals.org/content/42/8/e69.full.pdf+html">DSS</a>,
        <a href="http://bioinformatics.oxfordjournals.org/content/32/10/1446.short">DSS-single</a>,
	<a href="http://journal.frontiersin.org/article/10.3389/fgene.2014.00324/full">A review for DMR calling</a>.
    <li>Single-cell RNA-seq: 
	<a href="http://www.nature.com/nbt/journal/v32/n4/full/nbt.2859.html">Monocle</a>, 
        <a href="http://www.sciencedirect.com/science/article/pii/S0092867414004711">Wanderlust</a>,
	<a href="http://www.nature.com/nature/journal/v525/n7568/abs/nature14966.html">RaceID</a>,
	<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0844-5">MAST</a>,
        <a href="http://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0927-y">
	A review for design and analysis of scRNA-seq</a>.

</ol>


<center><hr size="5" width="100%"></center>
<p><a name="projects"><h3>Final project:</h3></a><p>

<p><u><h4>Possible topics</h4></u><p>

The final project could be (but not limited to) exploratory analysis,
statistical modeling, or analytical software development for 
any type of genomic data. Some ideas of projects include:
<ul>
<li>Exploring the DNA sequence content (e.g., GC content and dinucleotide frequencies) 
or gene annotations in different species.
<li>Exploring the relationship of protein binding/histone modifications and gene expression.
<li>Comparing gene expression microarrays and RNA-seq.
<li>A new statistical method for data normalization or differential expression detection for RNA-seq.
<li>Developing an R function to characterize a list of genomic intervals.
<li>New clustering method for single cell RNA-seq.
</ul>
Talk to instructor and/or TA for ideas or for helps in getting proper dataset.<p>

<p><u><h4>Requirement for the report</h4></u>

Students need to submit a short report for the final project.
The report needs to be a single docx or pdf file with file name as NAME_BIOS555_finalproject.docx. 
Here NAME is your full name. The R codes should be submitted as appendix, 
but need to be in the same file. 

The report should contain following sections: 
<ul>
<li><b>Introduction</b>: briefly introduce the scientific background and motivation.
<li><b>Method</b>: methods used for data analysis.
<li><b>Results</b>: data analysis results.
<li><b>Conclusion</b>: briefly summarize the findings from the data analysis. 
</ul>
Use tables and figures if needed. Make sure to  properly number the tables and figures,
and use captions. 

The report should NOT exceed 6 pages with single spacing, 11 point, 1 inch margin. 
R codes are not counted toward page limit. There is no minimum page requirement. 
<p>

<p><u><h4>Submit the report</h4></u>

The report is due on <b>Oct 22, 2021</b> at midnight, four days after the class ends.  
Please send the report (as a single attachment) to the instructor and cc the TA. 
<p style="color:red;">
 <b> NOTE: Due to the policy change in 2021, I MUST enter the students' grades in OPUS by 10/26.
   So I will not take project not submitted by the deadline. I appreciate if you can submit the project earlier,
   which will relieve my stress to grade 20+ final projects in a few days. 
</b>
  
<p><p><p><p>



<center><hr size="5" width="100%"></center>


<p><a name="class"><h3>Class schedule</h3> </a>
<table cellpadding=3 cellspacing=3 border=1>
<tr>
<td><b>Date</b></td>
<td><b>Lecture Title</b></td>
<td><b>Description</b></td>
<td><b>Homework</b></td>
<td><b>Suggested Reading</b></td>
</tr>

<tr>
<td>8/25 (Wed)</td>
<td><b>Lecture 1: Introduction</b> [<a href="intro.pdf">PDF</a>]</td>
<td>Brief introduction of molecular biology, high-throughput experiments,
R and Bioconductor.</td>
<td></td>
<td>Wikipedia pages for gene, genome, microarray and sequencing.</td> 
</tr>

<!-- <td>8/19 (W)</td>
<td>Labor day, no class</td>
<td></td>
<td></td>
<td></td>
</tr>
-->


<td>8/30 (Mon)</td>
<td><b>Lab 1: Simple genomic analysis using R</b> 
[<a href="lab1-intro.pdf">PDF</a>, <a href="lab1-intro.R">R</a>]
<td>Exploratory analysis of human refseq genes.</td>
<td><a href="homework1.pdf">Homework1</a></td>
<td></td>
</tr>


<td>9/1 (Wed)</td>
<td><b>Lecture 2: Gene expression microarray I</b>
[<a href="GE1.pdf">PDF</a>]</td>
<td>Experimental procedures and data pre-processing methods
for Gene expression microarrays. </td>
<td></td>
<td>The microarray review article, RMA and GCRMA papers.</td>
</tr>


<td>9/6 (Mon)</td>
<td>Labor day, no class</td>
<td></td>
<td></td>
<td></td>
</tr>

<td>9/8 (Wed)</td>
<td><b>Lecture 3: Gene expression microarray II, tiling arrays.</b>
[<a href="GE2.pdf">PDF</a>]</td>
<td>Differential expression from GE arrays. Biological and technical artifacts:
  batch effects, cell type mixture..</td>
<td></td>
<td>SAM and Limma papers.</td>
</tr>

<td>9/13 (Mon)</td>
<td><b>Lab 2: Analyzing gene expression array data from MAQC.</b>
[<a href="lab-GE.pdf">PDF</a>][<a href="lab-GE.R">R</a>]</td>
<td>Using gene expression microarrays generated by MAQC project,
we will explore data produced from different array designs and
compare to the gold standard. The gold standard Taqman data can be
found at <a href="Taqman_expr.txt">here</a>.</td>
<td><a href="homework2.pdf">Homework2</a></td>
<td></td>
</tr>

<td>9/15 (Wed)</td>
<td><b>Lecture 4: Handling genome data using Bioconductor I</b>
[<a href="biostrings.pdf">PDF</a>][<a href="biostrings.R">R</a>]</td>
<td>Introduce Biostrings and BSgenome Bioconductor packages.</td>
<td></td>
<td><a href="lawrencePLoSCompBio13.pdf">PLoS CB paper</a>,
Package Vignettes for Biostrings and BSgenome.</td>
</tr>

<td>9/29 (Mon)</td>
<td><b>Lecture 5: Handling genome data using Bioconductor II</b>
[<a href="genomicRanges.pdf">PDF</a>][<a href="genomicRanges.R">R</a>]</td>
<td>Introduce GenomicRanges and GenomicFeatures Bioconductor packages.
<td></td>
<td><a href="lawrencePLoSCompBio13.pdf">PLoS CB paper</a>, 
  Package Vignettes for GenomicRanges and GenomicFeatures.</td>
</tr>

<td>9/22 (Wed) </td>
<td><b>Lab 3: Analyzing human genome</b>
[<a href="lab3.pdf">PDF</a>][<a href="lab3.R">R</a>]</td>
<td>Study the sequence composition of human genome. Look at overlaps of CpG islands and gene promoters. 
<ul>
<li>CpG island for Human hg19: <a href="hg19_CGI.txt">hg19_CGI.txt</a>.
<li>TxDb object for Human hg19 genes: <a href="hg19_knownGenes.sqlite">hg19_knownGenes.sqlite</a>.
</ul>
</td>
<td><a href="homework3.pdf">Homework3</a></td>
<td></td>
</tr>

<td>9/27 (Mon)</td>
<td><b>Lecture 6: Introduction to second generation sequencing</b>
[<a href="sgseq.pdf">PDF</a>]</td>
<td>Introduce second generation sequencing technologies
and software for alignment, variant calling and visualization.</td>
<td></td>
<td></td>
</tr>

<td>9/29 (Wed)</td>
<td><b>Lecture 7: RNA-seq</b>
[<a href="RNAseq.pdf">PDF</a>]</td>
<td>Experimental procedure and data analysis for RNA-seq data. 
Normalization and differential expression detection. 
DEseq and edgeR Bioconductor package.</td>
<td></td>
<td>edgeR, DESeq, and cufflink papers</td>
</tr>



<td>10/4 (Mon)</td>
<td><b>Lecture 8: ChIP-seq</b>
[<a href="ChIPseq.pdf">PDF</a>]</td>
<td>Experimental procedure of ChIP-seq.
Peak calling methods. Comparison of multiple ChIP-seq.
Joint analysis of ChIP-seq and RNA-seq.</td>
<td></td>
<td>MACS and PeakSeq papers</td>
</tr>

<td>10/6 (Wed)</td>
<td><b>Lab 4: Handling second generation sequencing data, RNA- and ChIP-seq analyses</b>
[<a href="sgseq-lab.pdf">PDF</a>][<a href="sgseq-lab.R">R</a>]</td>
<td>This lab will practice the materials covered in three lectures.
(1) sequence alignment and manipulation.
(2) Joint analysis of ChIP-seq and RNA-seq.
(3) RNA-seq analysis using DEseq and edgeR.
<ul>
<li><a href="sgseqLab-data.zip">Data used in the lab</a>. 
<li><a href="knowngene.hg18.sqlite">TxDb object for hg18 known gene</a>. 
</td>
<td><a href="homework4.pdf">Homework4</a></td>
<td></td>
</tr>

<td>10/11 (Mon)</td>
<td>Fall break, no class</td>
<td></td>
<td></td>
<td></td>
</tr>


<td>10/13 (Wed)</td>
<td><b>Lecture 9: Bisulfite sequencing</b>
[<a href="bsseq.pdf">PDF</a>]</td>
<td>Experimental procedure of bisulfite sequencing.
Differential methylation. DNA methylation and protein binding.</td>
<td></td>
<td>BSmooth and DSS papers</td>
</tr>

<td>10/18 (Mon)</td>
<td><b>Lecture 10: Single-cell sequencing</b>
[<a href="scSeq.pdf">PDF</a>]</td>
<td>Briefly introduce single-cell sequencing technologies and data analysis, 
with emphasis on single-cell RNA sequencing (scRNA-seq). </td>
<td></td>
<td>Monocle, Wanderlust, RaceID, and MAST papers</td>
</tr>


</body></html>
